src/store/middleware.js

```diff
@@ -0,0 +1,31 @@
+import { RECEIVE_MESSAGE, setActiveChat } from "./Chat/index";
+
+
+export const logger = store => next => action => {
+  console.group(action.type)
+  console.info('dispatching', action)
+  let result = next(action)
+  console.log('next state', store.getState())
+  console.groupEnd(action.type)
+  return result
```

Comment:
just do `return next(action)`. No need for the `result` variable.

Reply:
`console.log`'s for days, whats `group` and `groupEnd` do?

Reply:
groups console logs by foldeable group. v. cool. this is not going to be implemented; it's just a logger and will be opt in for the dev but not committed to use for everyone.